core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -142,6 +143,21 @@ public B reduce(List<B> buckets, ReduceContext context) {
         }
 
         @Override
+        public final XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            doXContentKey(builder);
+            builder.field(CommonFields.DOC_COUNT, getDocCount());
+            if (showDocCountError) {
+                builder.field(InternalTerms.DOC_COUNT_ERROR_UPPER_BOUND_FIELD_NAME, getDocCountError());
+            }
+            aggregations.toXContentInternal(builder, params);
+            builder.endObject();
+            return builder;
+        }
+
+        protected abstract XContentBuilder doXContentKey(XContentBuilder builder) throws IOException;
```

Comment:
My comment was more about name coherency than pure naming. I'm fine with both doXContentKey or keyToXContent with a preference for the later.